scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -143,9 +145,11 @@ def _handle_encoding(
             body = self._decode(body, encoding, max_size)
         return body, to_keep
 
+    @staticmethod
     def _split_encodings(
-        self, content_encoding: list[bytes]
+        content_encoding: list[bytes],
     ) -> tuple[list[bytes], list[bytes]]:
+        supported_encodings = {*ACCEPTED_ENCODINGS, b"x-gzip"}
```

Comment:
I wonder if we should do something to make it easier to subclass the middleware to extend support for additional encodings, and hopefully make the implementation slightly cleaner that the same time.

e.g. Define `HttpCompressionMiddleware.DECODERS` as:

```python
DECODERS = {
    "br": unbrotli,
    "deflate": inflate,
    "gzip": gunzip,
    "x-gzip": gunzip,
    "zstd": unzstd,
}

// …

body = self.DECODERS[encoding](body, max_size=max_size)

// …

if encoding not in self.DECODERS:
```

Note that I am also suggesting making decoding functions public, to enable easier mapping of new keywords, e.g. `"x-deflate": inflate`.

No strong opinion on this, though. It’s rather out of scope here. And I am not sure if making those functions public is a good idea.

Reply:
I recall suggesting a similar thing to @wRAR :) But it seems leaving it out of scope for this PR is better. It doesn't seem this PR adds any additional obstacles to implement it later.